modules/services/monitoring/prometheus-node-exporter.nix

```diff
@@ -0,0 +1,81 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    concatStringsSep
+    escapeShellArgs
+    getExe
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+  ;
+
+  cfg = config.services.prometheus-node-exporter;
+in {
+  options = {
+    services.prometheus-node-exporter = {
+      enable = mkEnableOption "Prometheus Node exporter";
+
+      package = mkPackageOption pkgs "prometheus-node-exporter" { };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = ":9100";
+        description = ''
+          Listener address and port where Node exporter exposes its HTTP interface
+        '';
+      };
+
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "--log.level=debug" ];
+        description = ''
+          Extra commandline options to pass to the Node exporter executable.
+        '';
+      };
+
+      enabledCollectors = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = ''
+          Collectors to enable in addition to the ones that are [enabled by default](https://github.com/prometheus/node_exporter#enabled-by-default).
+        '';
+      };
+
+      disabledCollectors = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "boottime" ];
+        description = ''
+          Collectors to disable from the list of collectors that are [enabled by default](https://github.com/prometheus/node_exporter#enabled-by-default).
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    launchd.daemons.prometheus-node-exporter = {
```

Comment:
I believe this will run as `root`. We need `user` and `group` from the NixOS module, and should default to something with a `_` in front to mark it as a role account. Unfortunately that means static allocation; take a look at `modules/services/hercules-ci-agent/default.nix` for an example. I can write the definitions for this if needed. cc @Enzime as well

Reply:
Added user and group.